lib/fabrik/BasePlatformManager.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const CONST = require('../constants');
+
+class BasePlatformManager {
+  constructor(guid, context) {
+    this.guid = guid;
+    this.context = context;
+  }
+
+  get platform() {
+    return this.context.platform;
+  }
+  preInstanceProvisionOperations(options) {
```

Comment:
similar to this void method do you think you have missed out `postInstanceDeleteOperations`, `preInstanceUpdateOperations` methods?

Reply:
Not doing any operation in postInstanceDeleteOperations, preInstanceUpdateOperations.

Reply:
Let's keep methods only which are currently used. We can add later if anything is needed.